hashes/src/sha256.rs

```diff
@@ -173,6 +174,7 @@ impl Midstate {
     ///
     /// Computes non-finalized hash of `sha256(tag) || sha256(tag)` for use in
     /// [`sha256t`](super::sha256t). It's provided for use with [`sha256t`](crate::sha256t).
+    #[must_use]
```

Comment:
What is the lint that is suggesting this? Even for clippy this seems dumb.

Reply:
Not sure off the top of my head, its one of the pedantic ones. I'll post it when I'm re-working this.

Reply:
There is a special place in hell for people trying to get zero clippy messages by making some non-sense change. At Blockstream a new hire tried to do this in our codebase and his patch would have introduced at least one bug.

Not saying using clippy to get pointed at potential optimizations doesn't make sense. But it's just not needed to silent all of them because many of them are spurious.

Reply:
> There is a special place in hell for people trying to get zero clippy messages

And, lolz, I guess I reserved my entry ticket already with this PR, I'll try not to use it :)